docs/apigateway.md

```diff
@@ -0,0 +1,110 @@
+
+Let's say you want to expose a set of actions for a book club for your friends.
+You have a series of actions to implement your backend for the book club:
+
+| action | http method | description |
+| ----------- | ----------- | ------------ |
+| getBooks    | GET | get book details  |
+| postBooks   | POST | adds a book |
+| putBooks    | PUT | updates book details |
+| deleteBooks | DELETE | deletes a book |
+
+
+Lets create an API for the Book club `Book Club` on basePath `/club` to manage the `books` resources.
+```
+$ wsk api-experimental create -n "Book Club" /club /books get getBooks
+$ wsk api-experimental create /club /books post postBooks
+$ wsk api-experimental create /club /books put putBooks
+$ wsk api-experimental create /club /books delete deleteBooks
+```
+
+Notice that the first action exposed with basepath `/club` gets gets the API label with name `Book Club` any other actions exposed under `/club` will be associated with `Book Club`
+
+Lets list all the actions that we just exposed.
+
+```
+$ wsk api-experimental list
+ok: apis
+Action                   Verb          API Name        URL
+getBooks                   get         Book Club       https://2ef15285-gws.api-gw.mybluemix.net/club/books
+postBooks                 post         Book Club       https://2ef15285-gws.api-gw.mybluemix.net/club/books
+putBooks                   put         Book Club       https://2ef15285-gws.api-gw.mybluemix.net/club/books
+deleteBooks             delete         Book Club       https://2ef15285-gws.api-gw.mybluemix.net/club/books
+```
+
+Now just for fun lets add a new book `JavaScript: The Good Parts` with a HTTP __POST__
+```
+$  curl -X POST -d '{"name":"JavaScript: The Good Parts", "isbn":"978-0596517748"}' https://2ef15285-gws.api-gw.mybluemix.net/club/books
+```
+```
+{
+  "result": "success"
+}
+```
+
+Lets get a list of Books using our action `getBooks` via HTTP __GET__
+```
+$  curl -X GET https://2ef15285-gws.api-gw.mybluemix.net/club/books
+```
+```
+{
+  "result": [{"name":"JavaScript: The Good Parts", "isbn":"978-0596517748"}]
+}
+```
+
+You can delete all the exposed URL using basepath `/club` or API name label `"Book Club"`:
+```
+$ wsk api-experimental delete /club
+```
+```
+ok: deleted API /club
+```
+
+**Note**: The feature is being offered as experimental this will allow users to try the feature and
+provide feedback, the following is a list of known limitations and feedback already received:
+  - No ability to control HTTP access control for  Cross-Origin Resource Sharing (CORS).
+  - Only Content Type `application/json` is supported for request and response.
+  - No programatic way to control the response from the OpenWhis Action.
+  - All OpenWhisk actions are expose via public access, no ability to configure a custom API key.
+  - Path parameters are not supported, only query parameter and request body.
+  - API once created with no specific name, the name will be the basepath and this can't be updated.
```

Comment:
-> `If the API is created without an API name, the name will be the base path and this cannot be changed`